### DISCUSSION TITLE
are native detour not called in single player?

### DISCUSSION THREAD

**Discussion by lonelybud** [YimMenu/YimMenu#2449] [Q&A]
load menu in single player (offline)

comment out `ENTITY::SET_ENTITY_HEALTH(entity, health, p2);` in  `src/native_hooks/all_scripts.hpp`. Now take damage. Then click `Self > Heal` inside menu. See that health boost ups.

**Comment by Yimura** [comment]
What exactly is the problem? Native detours don't apply for natives ran in the menu.
Unless I'm misunderstanding...

**Reply by lonelybud** [reply]
you are right the native detour are called. I just verified that `SET_CURRENT_PED_WEAPON` was called after one die in SP offline.

**Comment by lonelybud** [comment]
is this hook `src\hooks\script\init_native_tables.cpp` called once. When i inject meny first time it seems to be called. When i unload and load menu again, it does not seem to be called.
